code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -66,9 +66,10 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 		//if assert_gas() was called only to read from the gas.
 	//By removing empty gases, processing speed is increased.
 /datum/gas_mixture/proc/garbage_collect(list/tocheck)
-	var/list/cached_gases = gases
-	for(var/id in (tocheck || cached_gases))
-		if(cached_gases[id][MOLES] <= 0 && cached_gases[id][ARCHIVE] <= 0)
+	var/list/cached_gases = tocheck || gases

```

Comment:
this does not maintain behavior. `tocheck` is not a proper gas list, it's just a list of paths.

Reply:
huh, nvm then